CoreFoundation/String.subproj/CFStringUtilities.c

```diff
@@ -603,9 +603,8 @@ CF_PRIVATE CFComparisonResult _CFCompareStringsWithLocale(CFStringInlineBuffer *
     if (range2.location > 0) {
 	range2.location = __extendLocationBackward(range2.location - 1, str2, nonBaseBMP, punctBMP);
     }
-    
+
 #if TARGET_OS_MAC || TARGET_OS_WIN32 || TARGET_OS_LINUX
-#if TARGET_OS_MAC
```

Comment:
This exposes the variable collator at l616, which is still hidden behind an #if tag above (l577), so this doesn't build properly on other platforms.

Reply:
Thanks for catching this. Do you think the the `#if TARGET_OS_MAC || TARGET_OS_WIN32 || TARGET_OS_LINUX` could be removed entirely and the code would work on the BSD platforms? btw which platforms are you testing on?

Reply:
Thanks, that builds cleanly now.

The platform-specific branches could be removed --  I did a rush check adding TARGET_OS_BSD on every appearance of TARGET_OS_LINUX atop this commit -- but they'd need to be removed in more places than you are changing here, so perhaps that should be done in a separate PR.

However, I haven't run any of the tests; right now I am just checking to ensure things build properly on OpenBSD without error.

Reply:
Thanks for checking, yes probably best to support other platforms in a future PR.